### PR TITLE
[RideHome] Add extractor

### PR DESCRIPTION
**Pull request by SirElderling**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Base extractor for `Techmeme Ride Home` podcast episodes.

_Note: From time to time, the request to download the initial website fails with HTTP Error 503: Service Unavailable._ 

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [X] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [X] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [X] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [X] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [X] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

**Comment by seproDev** [comment]
Do you think it would be better to make a generic art19 extractor?
A fair number of other sites like Nebula (https://github.com/yt-dlp/yt-dlp/issues/8838) also uses art19 for their podcast so having a generic metadata extractor would be helpful. 

The extractor could then match both direct mp3 links `https://rss.art19.com/episodes/7fd06487-59c8-4f3d-b6c5-8d64b114522f.mp3`.
As well as embeds `https://www.art19.com/shows/techmeme-ridehome/episodes/540e5493-9fe6-4c14-a488-dc508d8794b2/embed?theme=light-custom`

For this site specifically, the generic extractor should then be enough.

**Comment by seproDev** [comment]
Do you want to rework this returning URL results to Art19IE? https://github.com/yt-dlp/yt-dlp/commit/999ea80beb053491089d256104c4188aced3110f

**Comment by SirElderling** [comment]
> Do you want to rework this returning URL results to Art19IE? [999ea80](https://github.com/yt-dlp/yt-dlp/commit/999ea80beb053491089d256104c4188aced3110f)

Hi @seproDev . Apologies for the late response :) Yes, I'll rework this extractor to leverage ``Art19IE``.

File: yt_dlp/extractor/ridehome.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            get_elements_html_by_class(
+                'iframeContainer', webpage), (..., {extract_attributes}, 'data-src'))
+        if not containers:
+            raise ExtractorError('Unable to extract any media containers from webpage')
+
+        # couldn't find an example with multiple containers. This is just a safeguard.
+        if len(containers) > 1:
+            return self.playlist_result(self._entries(containers), article_id)
+
+        return self.url_result(containers[0], Art19IE)
```
Comment (seproDev) [inline yt_dlp/extractor/ridehome.py:74]: How about:
```suggestion
        urls = traverse_obj(
            get_elements_html_by_class('iframeContainer', webpage),
            (..., {extract_attributes}, lambda k, v: k == 'data-src' and Art19IE.suitable(v)))

        return self.playlist_result(self._entries(urls), article_id)
```
That way, we prevent iframes from getting picked up that don't actually contain art19 URLs.

File: yt_dlp/extractor/ridehome.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            'release_date': '20220108',
+            'release_timestamp': 1641672000,
+            'duration': 2789.38122,
+            'thumbnail': r're:^https?://content\.production\.cdn\.art19\.com/images/.*\.jpeg$'
+        }
+    }]
+
+    def _entries(self, containers):
+        for container in containers:
+            yield self.url_result(container, Art19IE)
```
Comment (seproDev) [inline yt_dlp/extractor/ridehome.py:58]: Maybe these names are a bit clearer
```suggestion
    def _entries(self, urls):
        for art19_url in urls:
            yield self.url_result(art19_url, Art19IE)
```

File: yt_dlp/extractor/ridehome.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+from ..utils import (
+    ExtractorError,
+    extract_attributes,
+    get_elements_html_by_class,
+    traverse_obj,
+)
+
+
+class RideHomeIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?ridehome\.info/show/techmeme-ride-home/(?P<id>[\w-]+)(?:/|$)'
```
Comment (seproDev) [inline yt_dlp/extractor/ridehome.py:13]: Other shows exist
```suggestion
    _VALID_URL = r'https?://(?:www\.)?ridehome\.info/show/[\w-]+/(?P<id>[\w-]+)/?(?:$|[?#])'
```

File: yt_dlp/extractor/ridehome.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
+from .common import InfoExtractor
+from .art19 import Art19IE
+
+from ..utils import (
+    ExtractorError,
+    extract_attributes,
+    get_elements_html_by_class,
+    traverse_obj,
+)
```
Comment (seproDev) [inline yt_dlp/extractor/ridehome.py:9]:
```suggestion
from .art19 import Art19IE
from .common import InfoExtractor
from ..utils import extract_attributes, get_elements_html_by_class
from ..utils.traversal import traverse_obj
```

**Comment by SirElderling** [comment]
I noticed that this show ``https://www.ridehome.info/show/kottke-ride-home/`` doesn't use ``Art19``. Something to improve upon in a new PR.

**Review by bashonly — APPROVED** [review]

**Comment by SirElderling** [comment]
Should we exclude the ``https://www.ridehome.info/show/kottke-ride-home/`` from matching the `` _VALID_URL``?
```python
 _VALID_URL = r'https?://(?:www\.)?ridehome\.info/show/(?!kottke)[\w-]+/(?P<id>[\w-]+)/?(?:$|[?#])'
```

**Comment by bashonly** [comment]
> Should we exclude the `https://www.ridehome.info/show/kottke-ride-home/ from matching` the  `_VALID_URL`?

Since it doesn't work with the generic extractor either, I think it's OK to leave the `_VALID_URL` as-is. Support can be added in another PR as you said above

**Pull request merged**